R/exportDataQuality.R

```diff
@@ -0,0 +1,92 @@
+
+exportDataQuality <- function(rcon, prefix,
+                                     ...){
+  UseMethod("exportDataQuality")
+}
+
+exportDataQuality.redcapApiConnection <- function(rcon, prefix,
+                                                       ..., 
+                                                       error_handling = getOption("redcap_error_handling"), 
+                                                       config = list(), 
+                                                       api_param = list()){
+  ###################################################################
+  # Argument Validation                                          ####
+  
+  coll <- checkmate::makeAssertCollection()
+  
+  checkmate::assert_class(x = rcon, 
+                          classes = "redcapApiConnection", 
+                          add = coll)
+  
+  checkmate::assert_class(x = prefix, 
+                          classes = "character", 
+                          add = coll)
+  
+  error_handling <- checkmate::matchArg(x = error_handling, 
+                                        choices = c("null", "error"),
+                                        .var.name = "error_handling",
+                                        add = coll)
+  
+  checkmate::assert_list(x = config, 
+                         names = "named", 
+                         add = coll)
+  
+  checkmate::assert_list(x = api_param, 
+                         names = "named", 
+                         add = coll)
+  
+  checkmate::reportAssertions(coll)
+  
+  ###################################################################
+  # Build the query list                                         ####
+  
+  url <- paste0("https://redcap.vanderbilt.edu/api/?prefix=", prefix, "&page=export&type=module&NOAUTH&pid=", rcon$projectInformation()$project_id)
+  
+  formData <- list(token = rcon$token)
+  
+  response <- httr::POST(url, body = formData, encode = "form")
+  tryCatch({
+    result <- httr::content(response, type = 'application/json')
+    
+    for(j in 1:length(result)){i=result[[j]];if(is.null(i$resolutions)){result[[j]]$resolutions=list()}}
+    result <- as.data.frame(do.call(rbind, result))
```

Comment:
The rest of the library sticks with CSV coming back. Cole and I did timing tests and found no benefit to JSON, primarily due to these two lines. This will work, but it's worth considering csv for consistency. Also, the csv path has protections now against miscoded data.

Reply:
Attempted to pull CSV and we found that we are stuck with json for now.